filesystem/android_super.ksy

```diff
@@ -0,0 +1,196 @@
+  application: Android
+  file-extension: img
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  endian: le
+
+doc: |
+  The metadata stored by Android at the beginning of a "super" partition, which
+  is what it calls a disk partition that holds one or more Dynamic Partitions.
+  Dynamic Partitions do more or less the same thing that LVM does on Linux,
+  allowing Android to map ranges of non-contiguous extents to a single logical
+  device. This metadata holds that mapping.
+
+doc-ref:
+  - https://source.android.com/devices/tech/ota/dynamic_partitions/implement
+  - https://android.googlesource.com/platform/system/core/+/master/fs_mgr/liblp/include/liblp/metadata_format.h
+
+seq:
+  - id: padding
+    size: 0x1000
+    type: dummy
+
+  - id: primary_geometry
+    size: 0x1000
+    type: geometry
+
+  - id: backup_geometry
+    type: geometry
+    size: 0x1000
+
+  - id: primary_metadata
+    size: primary_geometry.metadata_max_size
+    type: metadata
+    repeat: expr
+    repeat-expr: primary_geometry.metadata_slot_count
+
+  - id: backup_metadata
+    size: primary_geometry.metadata_max_size
+    type: metadata
+    repeat: expr
+    repeat-expr: primary_geometry.metadata_slot_count
+
+types:
+  dummy: {}
+
+  geometry:
+    seq:
+      - id: magic
+        contents: 'gDla'
+      - id: struct_size
+        type: u4
+      - id: checksum
+        size: 32
+      - id: metadata_max_size
+        type: u4
+      - id: metadata_slot_count
+        type: u4
+      - id: logical_block_size
+        type: u4
+
+  metadata:
+    seq:
+      - id: magic
+        contents: '0PLA'
+      - id: major_version
+        type: u2
+      - id: minor_version
+        type: u2
+      - id: header_size
+        type: u4
+      - id: header_checksum
+        size: 32
+      - id: tables_size
+        type: u4
+      - id: tables_checksum
+        size: 32
+      - id: partitions
+        type: table_descriptor(table_kind::partitions)
+      - id: extents
+        type: table_descriptor(table_kind::extents)
+      - id: groups
+        type: table_descriptor(table_kind::groups)
+      - id: block_devices
+        type: table_descriptor(table_kind::block_devices)
+
+    enums:
+      table_kind:
+        0: partitions
+        1: extents
+        2: groups
+        3: block_devices
+
+    types:
+      table_descriptor:
+        params:
+          - id: kind
+            type: u1
+            enum: table_kind
+
+        seq:
+          - id: offset
+            type: u4
+          - id: num_entries
+            type: u4
+          - id: entry_size
+            type: u4
+
+        instances:
+          table:
+            pos: _parent.header_size + offset
+            size: entry_size
+            type:
+              switch-on: kind
+              cases:
+                'table_kind::partitions': partition
+                'table_kind::extents': extent
+                'table_kind::groups': group
+                'table_kind::blocks': block_device
```

Comment:
Phew, it took me a while to debug this problem in the [Web IDE](https://ide.kaitai.io/), but apparently the compiler doesn't verify if the enum key exists. And `table_kind::blocks` enum value doesn't exist, it's named `block_devices` instead:

```suggestion
                'table_kind::block_devices': block_device
```

Reply:
Oops, my bad. Good catch!

Reply:
> Oops, my bad. Good catch!

In fact, not your bad at all 😐 The point is that the compiler should to catch these things, in the same way as when you reference a non-existing type or field.